account_asset_disposal/models/account_asset_asset.py

```diff
@@ -14,48 +15,152 @@ class AccountAssetAsset(models.Model):
     )
     disposal_date = fields.Date(string="Disposal date")
     disposal_move_id = fields.Many2one(
-        comodel_name='account.move', string="Disposal move")
+        comodel_name='account.move', string="Disposal move",
+    )
+
+    def _disposal_line_asset_prepare(self, date):
+        self.ensure_one()
+        return {
+            'name': _('Asset disposal'),
+            'journal_id': self.category_id.journal_id.id,
+            'account_id': self.category_id.account_asset_id.id,
+            'date': date,
+            'debit': 0.0,
+            'credit': self.value,
+        }
+
+    def _disposal_line_depreciation_prepare(self, date):
+        self.ensure_one()
+        loss_value = self.salvage_value + self.value_residual
+        depreciation_value = self.value - loss_value
+        return {
+            'name': _('Asset depreciation'),
+            'journal_id': self.category_id.journal_id.id,
+            'account_id': self.category_id.account_depreciation_id.id,
+            'date': date,
+            'debit': depreciation_value,
+            'credit': 0.0,
+        }
 
-    def get_disposal_date(self):
-        return fields.Date.context_today(self)
+    def _disposal_line_loss_prepare(self, date, loss_account):
+        self.ensure_one()
+        loss_value = self.salvage_value + self.value_residual
```

Comment:
I have some doubt with this, but they are beyond the scope of this module.

Reply:
It was validated by the customer.